File: swift/Sources/sdnx/Types/ObjectSchema.swift
import Foundation

public struct ObjectSchema: SchemaValue {
    public let type: String = "object"
    public var description: String?
    public var validators: [String: ValidatorInfo]?
    public let inner: Schema
    
    public init(inner: Schema, description: String? = nil, validators: [String: ValidatorInfo]? = nil) {
        self.inner = inner
        self.description = description
        self.validators = validators
    }
}

File: dotnet/Sdnx.Core/Types/ObjectSchema.cs
namespace Sdnx.Core
{
    public record ObjectSchema : SchemaValue
    {
        public required string Type { get; init; } = "object";
        public required Dictionary<string, SchemaValue> Inner { get; init; }
        public string? Description { get; init; }
        public Dictionary<string, Validator>? Validators { get; init; }
    }
}
